src/interface/meshblock_data.hpp

```diff
@@ -664,6 +659,14 @@ class MeshBlockData {
                                            vpack_types::VPackKey_t *key);
 };
 
+template <typename T, typename... Args>
+std::vector<Uid_t> UidIntersection(MeshBlockData<T> *mbd1, MeshBlockData<T> *mbd2,
+                                   Args &&...args) {
+  auto u1 = mbd1->GetVariableUIDs(std::forward<Args>(args)...);
+  auto u2 = mbd2->GetVariableUIDs(std::forward<Args>(args)...);
```

Comment:
Not that this matters much, but this code is identical between `MeshData` and `MeshBlockData`. I wonder if there's a way to reduce code duplication, either by templating on data type, or by having this version reference the `MeshBlockData` version?